master/buildbot/data/buildrequests.py

```diff
@@ -144,7 +168,16 @@ def get(self, resultSpec, kwargs):
             bsid=bsid,
             resultSpec=resultSpec)
         results = []
+        filters = resultSpec.popProperties() if hasattr(resultSpec, 'popProperties') else []
         for br in buildrequests:
+            if filters:
```

Comment:
looks like there is oportunity for factorization. like db2data.

Reply:
Agreed. Done in updated commit.